### PR TITLE
Added Laravel 9 Support

### PR DESCRIPTION
**Pull request by nasrulhazim**
I've added Laravel 9 support, together with Github Actions. The only caveat on this PR is, i'm dropping PHP 7.4 and Laravel 8 and lower support. Those versions should be rely on the other branches.

In Github Actions i've added 2 workflow:

1. Unit Test
2. PHP Code Style Fixer

### CODE REVIEW

**Comment by zerodahero** [comment]
Awesome!!! Thank you so much! Looks like this includes some of the changes in the 3.4 branch, which still have some open issues on, but as soon as those are resolved, we can release a 4.0 version.

**Comment by klimenttoshkov** [comment]
Zack,

When do you plan on releasing it? If it is going to take a while, I suggest to first release a version with L9 support, then update.

**Comment by NelinD** [comment]
@nasrulhazim Deprecated: ZeroDaHero\LaravelWorkflow\Events\AnnounceEvent implements the Serializable interface, which is deprecated. Implement __serialize() and __unserialize() instead (or in addition, if support for old PHP versions is necessary)

runtime [php 8.1.2]
laravel components [9]

https://www.php.net/manual/en/migration81.deprecated.php

**Comment by klimenttoshkov** [comment]
@NelinD  This should be addressed in the base event, which implements serialization features. And they are needed to be able to handle all events in queue.

**Comment by NelinD** [comment]
@klimenttoshkov 
This message triggers the package code.
I think it's better to make changes as part of this pull request.
Especially since it hasn't been accepted yet)

**Comment by nasrulhazim** [comment]
will try update the codes over the weekend. 😉

**Comment by nasrulhazim** [comment]
@zerodahero , i've update the codes to [fix the Serializable deprecation in PHP8.1](https://github.com/zerodahero/laravel-workflow/pull/63/commits/e83de20dedf51255c14babce86c70846ac568659). 

I've included backward compatibility as well to support PHP 7.4 and below.

Reference of fixes: https://php.watch/versions/8.1/serializable-deprecated

File: src/Events/BaseEvent.php [hunk 1]
```diff
@@ -16,6 +16,28 @@
  */
 abstract class BaseEvent extends Event implements Serializable
 {
+    public function __serialize(): array
```
Comment (NelinD) [inline src/Events/BaseEvent.php:19]: Maybe
```
    public function __serialize(): array
    {
        return [
            'base_event_class' => get_class($this),
            'subject' => $this->getSubject(),
            'marking' => $this->getMarking(),
            'transition' => $this->getTransition(),
            'workflow' => [
                'name' => $this->getWorkflowName(),
            ],
        ];
    }

    public function __unserialize(array $data): void
    {
        $this->subject = $data['subject'];
        $this->marking = $data['marking'];
        $this->transition = $data['transition'] ?? null;
        $workflowName = $data['workflow']['name'] ?? null;
        $this->workflow = Workflow::get($this->subject, $workflowName);
    }

    public function serialize()
    {
        return serialize($this->__serialize());
    }

    public function unserialize($serialized)
    {
        $this->__unserialize(unserialize($serialized));
    }
```

Comment (nasrulhazim) [reply src/Events/BaseEvent.php:19]: I've updated the code, updated as suggested, works good.

**Comment by NelinD** [comment]
@zerodahero When should we expect a release?

**Comment by zerodahero** [comment]
@NelinD likely in the next day or two. There's several other issues that are being tested as part of this release. If you're willing, you can help test the `develop` branch version of this package in the meantime.

**Comment by zerodahero** [comment]
@NelinD just released [v4.0.0](https://github.com/zerodahero/laravel-workflow/releases/tag/v4.0.0), please let me know if you run into any issue in the upgrade!

**Pull request merged**